azurerm/resource_arm_storage_account.go

```diff
@@ -548,6 +556,19 @@ func resourceArmStorageAccountCreate(d *schema.ResourceData, meta interface{}) e
 	log.Printf("[INFO] storage account %q ID: %q", storageAccountName, *account.ID)
 	d.SetId(*account.ID)
 
+	enableAdvanceThreatProtection := d.Get("enable_advanced_threat_protection").(bool)
+
+	advancedThreatProtectionSetting := security.AdvancedThreatProtectionSetting{
+		AdvancedThreatProtectionProperties: &security.AdvancedThreatProtectionProperties{
+			IsEnabled: &enableAdvanceThreatProtection,
```

Comment:
_minor_ you could do:
```suggestion
			IsEnabled: utils.bool(d.Get("enable_advanced_threat_protection").(bool)),
```

here instead.

Reply:
done